lib/ex_doc/formatter/html.ex

```diff
@@ -48,6 +48,8 @@ defmodule ExDoc.Formatter.HTML do
   defp generate_sidebar_items(modules, exceptions, protocols, output) do
     input = for node <- [%{id: "modules", value: modules}, %{id: "exceptions", value: exceptions}, %{id: "protocols", value: protocols}], !Enum.empty?(node.value), do: node
     content = Templates.sidebar_items_template(input)
+    # Safely reduce the size of the JS code
+    content = Regex.replace(~r/(\s|\n|\h|\v)/, content, "")
```

Comment:
We won't have space in those but this feels like such a hack. :D Do we have a really measurable gain? 

One alternative, just for sanity, is to be a bit more conservative and trim them only if the follow `(\:|\,)`. We can use one of the replace syntax to replace them but keep the leading character:

```iex
iex(2)> Regex.replace ~r/(\:|\,)(\s|\t|\R|\h|\v)+/, ~s("hello world": ["we are some strings"]), "\\1"
"\"hello world\":[\"we are some strings\"]"
```